src/AspNet.Security.OAuth.CiscoSpark/CiscoSparkAuthenticationHelper.cs

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (http://www.apache.org/licenses/LICENSE-2.0)
+ * See https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers
+ * for more information concerning the license and the contributors participating to this project.
+ */
+
+using JetBrains.Annotations;
+using Newtonsoft.Json.Linq;
+
+namespace AspNet.Security.OAuth.CiscoSpark {
+    /// <summary>
+    /// Contains static methods that allow to extract user's information from a <see cref="JObject"/>
+    /// instance retrieved from Spark after a successful authentication process.
+    /// </summary>
+    public static class CiscoSparkAuthenticationHelper {
+        /// <summary>
+        /// Gets the identifier corresponding to the authenticated user.
+        /// </summary>
+        public static string GetUserIdentifier([NotNull] JObject user) => user.Value<string>("id");
+
+        /// <summary>
+        /// Gets the login corresponding to the authenticated user.
+        /// </summary>
+        public static string GetUserName([NotNull] JObject user) => user.Value<string>("displayName");
+
+        /// <summary>
+        /// Gets the email corresponding to the authenticated user.
+        /// </summary>
+        public static string GetEmail([NotNull] JObject user)  => (string)((JArray)user["emails"]).First;
```

Comment:
This looks a bit fragile. Any chance you could opt for this kind of approach?
https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/blob/dev/src/AspNet.Security.OAuth.Spotify/SpotifyAuthenticationHelper.cs#L35